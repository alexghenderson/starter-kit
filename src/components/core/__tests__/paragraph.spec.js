import React from 'react';
import enzyme from 'enzyme';
import Paragraph from '../paragraph';

describe('Paragraph core component', () => {
  it('should render with default props', () => {
    const wrapper = enzyme.mount(<Paragraph>test-content</Paragraph>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with alternate tags', () => {
    const wrapper = enzyme.mount(<Paragraph tag="div">test-content</Paragraph>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render different sizes', () => {
    const wrapper = enzyme.mount(
      <React.Fragment>
        {['small', 'medium', 'large'].map(size => (
          <Paragraph size={size} key={size}>
            test-{size}
          </Paragraph>
        ))}
      </React.Fragment>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
