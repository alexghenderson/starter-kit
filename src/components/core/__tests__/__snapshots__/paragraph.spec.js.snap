// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph core component should render different sizes 1`] = `
Array [
  .emotion-0 {
  display: block;
  font-size: 0.875rem;
  -webkit-letter-spacing: -0.6px;
  -moz-letter-spacing: -0.6px;
  -ms-letter-spacing: -0.6px;
  letter-spacing: -0.6px;
  line-height: 1.42857;
  font-family: "Roboto",sans-serif;
}

<p
    className="emotion-0"
  >
    test-
    small
  </p>,
  .emotion-0 {
  display: block;
  font-size: 1rem;
  -webkit-letter-spacing: -0.8px;
  -moz-letter-spacing: -0.8px;
  -ms-letter-spacing: -0.8px;
  letter-spacing: -0.8px;
  line-height: 1.5;
  font-family: "Roboto",sans-serif;
}

<p
    className="emotion-0"
  >
    test-
    medium
  </p>,
  .emotion-0 {
  display: block;
  font-size: 1.25rem;
  -webkit-letter-spacing: -1px;
  -moz-letter-spacing: -1px;
  -ms-letter-spacing: -1px;
  letter-spacing: -1px;
  line-height: 1.6;
  font-family: "Roboto",sans-serif;
}

<p
    className="emotion-0"
  >
    test-
    large
  </p>,
]
`;

exports[`Paragraph core component should render with alternate tags 1`] = `
.emotion-0 {
  display: block;
  font-size: 1rem;
  -webkit-letter-spacing: -0.8px;
  -moz-letter-spacing: -0.8px;
  -ms-letter-spacing: -0.8px;
  letter-spacing: -0.8px;
  line-height: 1.5;
  font-family: "Roboto",sans-serif;
}

<div
  className="emotion-0"
>
  test-content
</div>
`;

exports[`Paragraph core component should render with default props 1`] = `
.emotion-0 {
  display: block;
  font-size: 1rem;
  -webkit-letter-spacing: -0.8px;
  -moz-letter-spacing: -0.8px;
  -ms-letter-spacing: -0.8px;
  letter-spacing: -0.8px;
  line-height: 1.5;
  font-family: "Roboto",sans-serif;
}

<p
  className="emotion-0"
>
  test-content
</p>
`;
